tests/unit/expire.tcl

```diff
@@ -280,47 +280,147 @@ start_server {tags {"expire" "external-ok"}} {
+            set ttl1 [r pexpiretime foo1]
+            set ttl2 [r pexpiretime foo2]
+            set ttl3 [r pexpiretime foo3]
+            set ttl4 [r pexpiretime foo4]
+            set ttl5 [r pexpiretime foo5]
+            set ttl6 [r pexpiretime foo6]
+            set ttl7 [r pexpiretime foo7]
+            set ttl8 [r pexpiretime foo8]
+            set ttl9 [r pexpiretime foo9]
+            set ttl10 [r pexpiretime foo10]
+            assert_equal "-2" [r pexpiretime foo11] ; # foo11 is gone
+            set ttl12 [r pexpiretime foo12]
+            set ttl13 [r pexpiretime foo13]
+            set ttl14 [r pexpiretime foo14]
+            set ttl15 [r pexpiretime foo15]
+            assert_equal "-1" [r pexpiretime foo16] ; # foo16 has no TTL
+            set ttl17 [r pexpiretime foo17]
+            set ttl18 [r pexpiretime foo18]
+
+            # Let some time pass and reload data from AOF
+            after 2000
+            r debug loadaof
+
+            # Assert that relative TTLs are roughly the same
+            assert_range [r ttl foo1] 90 98
+            assert_range [r ttl foo2] 90 98
+            assert_range [r ttl foo3] 90 98
+            assert_range [r ttl foo4] 90 98
+            assert_range [r ttl foo5] 90 98
+            assert_range [r ttl foo6] 90 98
+            assert_range [r ttl foo7] 90 98
+            assert_range [r ttl foo8] 90 98
+            assert_range [r ttl foo9] 90 98
+            assert_range [r ttl foo10] 90 98
+            assert_equal [r ttl foo11] "-2" ; # foo11 is gone
+            assert_range [r ttl foo12] 90 98
+            assert_range [r ttl foo13] 90 98
+            assert_range [r ttl foo14] 90 98
+            assert_range [r ttl foo15] 90 98
+            assert_equal [r ttl foo16] "-1" ; # foo16 has no TTL
+            assert_range [r ttl foo17] 90 98
+            assert_range [r ttl foo18] 90 98
+
+            # Assert that all keys have restored the same absolute TTLs from AOF
+            assert_equal [r pexpiretime foo1] $ttl1
+            assert_equal [r pexpiretime foo2] $ttl2
+            assert_equal [r pexpiretime foo3] $ttl3
+            assert_equal [r pexpiretime foo4] $ttl4
+            assert_equal [r pexpiretime foo5] $ttl5
+            assert_equal [r pexpiretime foo6] $ttl6
+            assert_equal [r pexpiretime foo7] $ttl7
+            assert_equal [r pexpiretime foo8] $ttl8
+            assert_equal [r pexpiretime foo9] $ttl9
+            assert_equal [r pexpiretime foo10] $ttl10
+            assert_equal [r pexpiretime foo11] "-2" ; # foo11 is gone
+            assert_equal [r pexpiretime foo12] $ttl12
+            assert_equal [r pexpiretime foo13] $ttl13
+            assert_equal [r pexpiretime foo14] $ttl14
+            assert_equal [r pexpiretime foo15] $ttl15
+            assert_equal [r pexpiretime foo16] "-1" ; # foo16 has no TTL
+            assert_equal [r pexpiretime foo17] $ttl17
+            assert_equal [r pexpiretime foo18] $ttl18
+        } {} {needs:debug}
```

Comment:
what was the trigger for this change? and what exactly did you do?
i see:
1. added a few more command like RESTORE 
2. added code that look into the AOF file to explicitly check what was propagated to it. (i think we have such a test elsewhere. or maybe just one that looks into the replication stream (which is now identical to the aof propagation so not sure we need both).
3. add a bunch of assertions to validate the EXPIRETIME after reload, not just the TTL. what's the value of doing that?
4. anything else?

Reply:
Bad merge, ignore it.